[CI] SSLReloadDuringStartupIntegTests.testReloadDuringStartup reproducibly fails in 7.x

This test fails in 7.x, for example:

```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.ssl.SSLReloadDuringStartupIntegTests.testReloadDuringStartup"   -Dtests.seed=AA29F22B124B878B   -Dtests.security.manager=true   -Dtests.locale=ar-EG   -Dtests.timezone=America/Port-au-Prince   -Dcompiler.java=14
```

The same seed does not fail in master.

The error looks like: 

```
 NoNodeAvailableException[None of the configured nodes are available: [{#transport#-1}{omeByUOxSW-wdvqjemUkDg}{127.0.0.1}{127.0.0.1:59635}]]
        at __randomizedtesting.SeedInfo.seed([AA29F22B124B878B:71FBCC66E5FC800F]:0)
        at org.elasticsearch.client.transport.TransportClientNodesService.ensureNodesAreAvailable(TransportClientNodesService.java:352)
        at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:248)
        at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:57)
        at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:396)
        at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:399)
```

But I think the real error comes from here:


```
  1> Caused by: java.security.SignatureException: Signature length not correct: got 512 but was expecting 256
  1>    at sun.security.rsa.RSASignature.engineVerify(RSASignature.java:212) ~[?:?]
  1>    at java.security.Signature$Delegate.engineVerify(Signature.java:1415) ~[?:?]
  1>    at java.security.Signature.verify(Signature.java:789) ~[?:?]
  1>    at sun.security.x509.X509CertImpl.verify(X509CertImpl.java:449) ~[?:?]
  1>    at sun.security.provider.certpath.BasicChecker.verifySignature(BasicChecker.java:166) ~[?:?]
  1>    at sun.security.provider.certpath.BasicChecker.check(BasicChecker.java:147) ~[?:?]
  1>    at sun.security.provider.certpath.PKIXMasterCertPathValidator.validate(PKIXMasterCertPathValidator.java:125) ~[?:?]
  1>    at sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:237) ~[?:?]
  1>    at sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:145) ~[?:?]
  1>    at sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(PKIXCertPathValidator.java:84) ~[?:?]
  1>    at java.security.cert.CertPathValidator.validate(CertPathValidator.java:309) ~[?:?]
  1>    at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:364) ~[?:?]
  1>    at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:275) ~[?:?]
  1>    at sun.security.validator.Validator.validate(Validator.java:264) ~[?:?]
  1>    at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:285) ~[?:?]
  1>    at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:144) ~[?:?]
  1>    at org.elasticsearch.common.ssl.DiagnosticTrustManager.checkServerTrusted(DiagnosticTrustManager.java:110) ~[elasticsearch-ssl-config-7.8.0-SNAPSHOT.jar:7.8.0-SNAPSHOT]
  1>    at sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1318) ~[?:?]
  1>    at sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1215) ~[?:?]
  1>    at sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1158) ~[?:?]
  1>    at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396) ~[?:?]
  1>    at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444) ~[?:?]
  1>    at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1260) ~[?:?]
  1>    at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1247) ~[?:?]
  1>    at java.security.AccessController.doPrivileged(AccessController.java:691) ~[?:?]
  1>    at sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:1192) ~[?:?]

```

https://gradle-enterprise.elastic.co/s/6d3jvqd2ro7wm
https://gradle-enterprise.elastic.co/s/yn5ia46i5ni4s
https://gradle-enterprise.elastic.co/s/7w4phbbuikw26

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLReloadDuringStartupIntegTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLReloadDuringStartupIntegTests.java
@@ -10,6 +10,7 @@ import org.elasticsearch.common.settings.MockSecureSettings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import org.elasticsearch.test.InternalTestCluster.RestartCallback;
 import org.elasticsearch.test.SecurityIntegTestCase;
 
@@ -21,6 +22,10 @@ import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 import java.util.concurrent.CountDownLatch;
 
+// transport clients do not support reloading of files so failures can occur if we allow the use
+// of transport clients in tests when we are changing files and a transport client is built from
+// the node we change the files on
+@ClusterScope(transportClientRatio = 0)
 public class SSLReloadDuringStartupIntegTests extends SecurityIntegTestCase {
 
     @Override
@@ -52,7 +57,6 @@ public class SSLReloadDuringStartupIntegTests extends SecurityIntegTestCase {
         return true;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/55524")
     public void testReloadDuringStartup() throws Exception {
         final String node = randomFrom(internalCluster().getNodeNames());
         final Environment env = internalCluster().getInstance(Environment.class, node);

